src/pocketmine/entity/Entity.php

```diff
@@ -1383,10 +1383,20 @@ protected function switchLevel(Level $targetLevel){
 		return true;
 	}
 
+	/**
+	 * @deprecated Use asPosition() instead getPosition()
+	 * 
+	 * @return Position
+	 */
 	public function getPosition(){
 		return new Position($this->x, $this->y, $this->z, $this->level);
 	}
 
+	/**
+	 * @deprecated Use asLocation() instead getLocation()
```

Comment:
@Muqsit there are actually many places in the PocketMine source that should use this method but did not do so.